rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +304,160 @@ def query_rules_and_serialize(
+        evm_accounting_aggregator: 'EVMAccountingAggregators',
+        events: Sequence[HistoryBaseEntry],
+) -> list[bool]:
+    """
+    For a list of events returns a list of the same length with boolean values where True
+    means that the event won't be affected by any accounting rule or processed in accounting
+    """
+    history_db = DBHistoryEvents(db)
+    with db.conn.read_ctx() as cursor:
+        # to know if an event will be processed or not in accounting we also need the related
+        # events since they might use callbacks or special treatments
+        related_events = history_db.get_history_events(
+            cursor=cursor,
+            filter_query=HistoryEventFilterQuery.make(
+                event_identifiers=list({event.event_identifier for event in events}),
+                order_by_rules=[('event_identifier', True), ('sequence_index', True)],
+            ),
+            has_premium=True,
+        )
+
+    query = 'SELECT COUNT(*) FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))'  # noqa: E501
+    bindings = [
+        (
+            event.event_type.serialize(),
+            event.event_subtype.serialize(),
+            NO_ACCOUNTING_COUNTERPARTY if (counterparty := getattr(event, 'counterparty', None)) is None else counterparty,  # noqa: E501
+            NO_ACCOUNTING_COUNTERPARTY,  # account for rules based only on type/subtype
+        ) for event in related_events
+    ]
+
+    callbacks = evm_accounting_aggregator.get_accounting_callbacks()
+    bindings_and_events_iterator = zip(bindings, related_events)
+    processed_events: dict[int, bool] = {}  # map of all the events related to the one requested to wether they will be processed in accounting or not  # noqa: E501
+    with db.conn.read_ctx() as cursor:
+        # index to keep the current event in the list of related events. It is used in the
+        # callbacks since we need to process events but we don't want to consume the current
+        # iterator
+        current_event_index = 0
+        for event_binding, event in bindings_and_events_iterator:
+            assert event.identifier is not None, 'We should only process events with identifiers'
+            if (cached_is_missing_rule := accountant.processable_events_cache.get(event.identifier)) is not None:  # noqa: E501
+                processed_events[event.identifier] = cached_is_missing_rule
+                current_event_index += 1
+                continue
+
+            if isinstance(event, EthStakingEvent) is True:
+                # staking events all have a process() function for accounting
+                processed_events[event.identifier] = False
+                current_event_index += 1
+                continue
+
+            # check the rule in the database
+            row = cursor.execute(query, event_binding).fetchone()
+            is_missing_rule = row[0] == 0
+            processed_events[event.identifier] = is_missing_rule
```

Comment:
The reason you don't stop here if this is False is to also populate the callback if existing? Why exactly?

Reply:
Because the event in addition to having an accounting rule could have a callback that affects events that come after and is not enough to check the accounting rule